docs: Enhance Bootstrap node deployment docs

Ebaneck opened the pull request:
**Component**:

'docs'
<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

See #2114 

**Summary**:

This PR adds the following: 
- Attempt to better explain  the bootstrap configuration file
- Adds minor troubleshooting guide
- Move the Prometheus volume provisioning step to post-install

**Acceptance criteria**: 

- Enhanced bootstrap documentation

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2114

bert-e commented:
# Hello ebaneck,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

NicolasT commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +50,37 @@ Configuration
+   .. code-block:: shell
 
-   - Explain the role of this config file and its values
+      NO_PROXY="localhost,127.0.0.1,10.10.0.0/16,192.168.0.0/16"
```
This seems confusing: not in YAML-format, and not using the same key name (caps vs. lower). Also, the YAML value must be a list (type), not a comma-separate string.

NicolasT commented on docs/installation/bootstrap.rst:
```diff
@@ -120,31 +141,17 @@ Bootstrap node.
 ^^^^^^^^^^^^^^^^^^^^
-Check if all :term:`Pods <Pod>` on the Bootstrap node are in the
-``Running`` state.
+- Check if all :term:`Pods <Pod>` on the Bootstrap node are in the
```
If we don't mention provisioning storage for Prometheus and such before this, then they *can't* be all in `Running` state.

NicolasT commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,54 @@
+
+Errors after restarting the Bootstrap node
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you restart the Bootstrap node and for some reason, some containers
```
Missing an `and` or something here.

NicolasT commented:
Please review @lucieleonard (or someone in your team).

thomasdanan commented:
Shouldn't we explain from where get the iso?

thomasdanan commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +50,41 @@ Configuration
 
+The ``archives`` field is a list of absolute paths to MetalK8s ISO files. When
+the bootstrap script is executed, these ISOs are automatically mounted and the
+system is configured to re-mount them automatically after a reboot.
```
Is the bootstrap script really mounting those iso when we run it? isn't it a manual step to perform? and when we execute the bootstrap script, we update system configuration to mount the iso automatically after a reboot?

Ebaneck replied:
Rephrased and fixed!

thomasdanan commented on docs/installation/bootstrap.rst:
```diff
@@ -187,16 +205,13 @@ Check if all :term:`Pods <Pod>` on the Bootstrap node are in the
    metalk8s-monitoring   prometheus-k8s-1                                 3/3       Running   1          3m        10.233.132.78   bootstrap   <none>
```
Ideally use the right version of K8S instead of 1.11.7

thomasdanan commented on docs/installation/bootstrap.rst:
```diff
@@ -187,16 +205,13 @@ Check if all :term:`Pods <Pod>` on the Bootstrap node are in the
+  :ref:`this procedure <quickstart-services-admin-ui>`.
 
-.. todo::
+- At this stage, the MetalK8s GUI should be up and ready for you to
```
even if prom and am PV have not been provisioned?

Ebaneck replied:
fixed

thomasdanan commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,54 @@
+to reset or change it.
+
+
+.. todo::
```
are you going to handle this in this PR?

Ebaneck replied:
I think the two last ones are not mission critical

gdemonet replied:
Troubleshooting for networking is indeed a large part and may be relegated to a future enhancement (worth opening a ticket to track this however, cc @thomasdanan).

For Salt logs, I think it would make sense to at least document it somewhere. Let's keep it for 2.4.2, but not this PR, which is already quite big.

thomasdanan reviewed (dismissed):
Nice job !!!
Just a few little comments + 

- Shouldn't we explain from where get the iso? and remind it should not be stored on /tmp?
- I thought we wanted to manage IP IP configuration through the bootstrap.yaml config file but it may not be available yet?

Ebaneck commented:
> Nice job !!!
> Just a few little comments +
> 
>     * Shouldn't we explain from where get the iso? and remind it should not be stored on /tmp?
> 

We could but it seems the built and released ISO's are strictly internal.
Also the ISO store path now reads it should be kept in the users home directory.

>     * I thought we wanted to manage IP IP configuration through the bootstrap.yaml config file but it may not be available yet?

I think this feature is not ready yet for 2.4.2 and btw, we have a small guide added in the `Installation` section which hints users on how to enable `IP-in-IP`.

NicolasT commented:
> Shouldn't we explain from where get the iso? and remind it should not be stored on /tmp?

I believe that's handled in @lucieleonard's PR.

Ebaneck commented:
> > Shouldn't we explain from where get the iso? and remind it should not be stored on /tmp?
> 
> I believe that's handled in @lucieleonard's PR.

True.

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,57 @@ Configuration
+
+      The ``controlPlane`` and ``workloadPlane`` entries are **mandatory**.
+      These values specify the range of IP addresses that will be attributed to
+      controlPlane and workloadPlane nodes.
```
```suggestion
      control plane and workload plane nodes.
```

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,57 @@ Configuration
+notation for the various fields.
+
+In production clusters, we advise users to choose a
+subnet range of atleast `/16` in order to have sufficient IP addresses keeping
```
`/16` is definitely **not** a requirement for control/workload plane networks: those are real IPs, so as long as you have all your nodes in them, it's enough (and 5 nodes fit in a `/28` if needed).
`/16` is however recommended for internal, virtual networks that are used for Pod and Service ClusterIPs.

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,57 @@ Configuration
+
+In production clusters, we advise users to choose a
+subnet range of atleast `/16` in order to have sufficient IP addresses keeping
+in mind that users infrastructure will grow in the near future.
```
`users infrastructure` doesn't mean much to me. The recommendation makes sense however, so maybe rephrasing it to something like:

> For production clusters, we advise users to anticipate future expansions and use sufficiently large networks for control plane and workload plane. A `/24` should be enough. Note that aggregation of subnet segments is not yet achievable with this configuration file.

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,57 @@ Configuration
+in mind that users infrastructure will grow in the near future.
+
+      The ``controlPlane`` and ``workloadPlane`` entries are **mandatory**.
+      These values specify the range of IP addresses that will be attributed to
```
Not `will be attributed`: those networks are the ranges of IP addresses used at the host level for each member of the cluster. It could also make sense to mention that **all** nodes in the cluster **must** have an interface connected to each of those networks (CP and WP) - this may be the same interface if we only use a single network range for both.

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,57 @@ Configuration
+              workloadPlane: 10.200.0.0/16
+
+      The ``pods`` and ``services`` fields at not mandatory but can be set
+      based on users requirements.
```
Maybe suggest why one would want to change these (`users requirements` is too vague IMHO). Something like:

> The ``pods`` and ``services`` fields are not mandatory, though can be changed to match the constraints of existing networking infrastructure (e.g. if all or part of these default subnets is already routed).

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,57 @@ Configuration
+      During installation, by default ``pods`` and ``services`` are set to the
+      following values below if omitted. This is done in order that pods and
+      services within the cluster obtain unique IP addresses and avoid any
+      conflicts.
```
The explanation lacks a useful disclaimer: those are internal, virtual networks. Nothing has to be done on the infrastructure side, as our cluster will handle configuration of the routing tables and allocation pools on its own (at the `iptables` level). We just need to make sure that they don't conflict with existing, routed networks.

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -160,43 +191,51 @@ Check if all :term:`Pods <Pod>` on the Bootstrap node are in the
+
+     The MetalK8s GUI will only become accessible when persistent storage
+     volumes for both Prometheus and Alertmanager have been successfully
+     provisioned.
```
That's not true anymore, UI is accessible, but Monitoring features will be missing.

gdemonet commented on docs/installation/post-install.rst:
```diff
@@ -1,2 +1,25 @@
+is not suitable for production installations. Refer to :ref:`volume-management`
+for more information on how to provision persistent storage.
+
+.. note::
+
+   When deploying using Vagrant, persistent volumes for Prometheus and
+   AlertManager are already provisioned.
```
Should this still be part of the Installation guide? cc @thomasdanan

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+This section highlights some of the common problems users face during and
+after a MetalK8s installation. If you do not find a solution to a problem you
+are facing, please reach out to us or create a
+`Github issue <https://github.com/scality/metalk8s/issues>`_.
```
Mention Scality support?

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If during a MetalK8s installation you encounter a failure and the console
+output does not provide sufficient information in order to pin-point the cause
+of failure, then re-run the installation with verbose arguments(--verbose).
```
```suggestion
of failure, then re-run the installation with the verbose flag (``--verbose``).
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+
+Errors after restarting the Bootstrap node
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you restart the Bootstrap node and for some reason, some containers
```
nitpick: `reboot` is more obvious than `restart`, maybe I'm wrong.

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+Persistent Volume Claim(PVC) stuck in "Pending" state
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If after provisioning a volume for a Pod say Prometheus and the PVC still
```
For clarity:

```suggestion
If after provisioning a volume for a Pod (e.g. Prometheus) and the PVC still
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If after provisioning a volume for a Pod say Prometheus and the PVC still
+hangs in a ``Pending`` state, then try checking the following.
```
Use an unordered list for the various checks

```suggestion
hangs in a ``Pending`` state, then try checking the following:

- Volumes have been provisioned and their status is ``Ready``: [...]
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+If after provisioning a volume for a Pod say Prometheus and the PVC still
+hangs in a ``Pending`` state, then try checking the following.
+
+Check that the volumes have been provisioned and it's status is ``Ready``:
```
Volume being an actual CRD, it could make sense to capitalize occurences of `Volumes`

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+
+   kubectl describe volume <volume name>
+
+Check that a corresponding Persistent volume exist:
```
```suggestion
Check that a corresponding PersistentVolume exist:
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+
+.. code-block:: shell
+
+   kubectl -n kube-system get pods
```
Better to look at the Deployment object I believe:

```suggestion
   kubectl -n kube-system get deployments storage-operator
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+
+   kubectl -n <namespace-name> logs -p  <pod name>
+
+Persistent Volume Claim(PVC) stuck in "Pending" state
```
Great troubleshooting suggestions, thanks!! I think we could improve those a bit by providing expected output, this would help users being more autonomous (if they have a hard time understanding Kubernetes)

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+
+.. code-block:: shell
+
+   kubectl -n metalk8s-ingress get pods
```
Ditto, Deployments give a simpler information, including the number of expected Pods compared to the actual number.

Ebaneck replied:
True... only that the deployments returns the following which is not sufficient i think :<

```
[root@bootstrap vagrant]# kubectl -n metalk8s-ingress get deployments
NAME                            READY   UP-TO-DATE   AVAILABLE   AGE
nginx-ingress-default-backend   1/1     1            1           4d18h

```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+
+.. code-block:: shell
+
+   kubectl -n metalk8s-ingress logs <nginx-ingress-pod-name>
```
Since Metal UI runs behind the control-plane Ingress, it would make sense to:

- provide a query to list all relevant Pods (using a label selector, e.g. `-l release=nginx-ingress-control-plane`)
- then show how to read the logs using the target Pod's name

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+
+Pod and Service CIDR conflicts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If after installation of a MetalK8s cluster you notice that Pod to pod
```
```suggestion
If after installation of a MetalK8s cluster you notice that Pod-to-Pod
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+Pod and Service CIDR conflicts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If after installation of a MetalK8s cluster you notice that Pod to pod
+communication has routing problems, perform the following.
```
Ditto, use a colon and list all checks in a unordered list

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,124 @@
+communication has routing problems, perform the following.
+
+Check that Pod network subnets do not conflict with host networks by verifying
+allocated pod IP addresses:
```
Maybe not the most straightforward approach. Let's add a first check on the actual network values:

```rst
- Check the configured values for the internal Pod and Service networks:

   .. code-block:: shell

      root@bootstrap # salt-call pillar.get metalk8s:networks

   Make sure the configured IP ranges (CIDR notation) do not conflict with your infrastructure.
```

After a check like this one, there shouldn't be a need to go through the `kubectl get pods` command. Going further may look into more advanced commands, such as `traceroute` or `tcpdump`. Not sure it's worth going too far in this iteration though.

gdemonet reviewed (changes requested):
Some comments, but good job overall, this doc is starting to look professional :)

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,60 @@ Configuration
+      following values below if omitted.
+
+      All nodes within the cluster **must** connect to both the control plane
+      and workload networks. If the same network range is chosen for both the
```
```suggestion
      and workload plane networks. If the same network range is chosen for both the
```

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,60 @@ Configuration
+
+      All nodes within the cluster **must** connect to both the control plane
+      and workload networks. If the same network range is chosen for both the
+      control plane and workload plane networks then the same interface maybe
```
```suggestion
      control plane and workload plane networks then the same interface may be
```

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,60 @@ Configuration
+      These values specify the range of IP addresses that will be attributed to
+      control plane and workload plane nodes.
+
+      .. code-block:: shell
```
For this one, and others below, why not use `.. code-block:: yaml`?

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,60 @@ Configuration
+              controlPlane: 10.200.1.0/28
+              workloadPlane: 10.200.1.0/28
+
+      The ``pods`` and ``services`` fields are not mandatory, though can be
+      changed to match the constraints of existing networking infrastructure
+      (for example, if all or part of these default subnets is already routed).
```
:+1:

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -47,21 +52,60 @@ Configuration
+      All nodes within the cluster **must** connect to both the control plane
+      and workload networks. If the same network range is chosen for both the
+      control plane and workload plane networks then the same interface maybe
+      used.
```
I think this paragraph should be moved just after the `controlPlane`/`workloadPlane` example, for clarity. Here, we're talking about `pods`/`services`, so it's a bit confusing.

Ebaneck replied:
true

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -160,43 +194,51 @@ Check if all :term:`Pods <Pod>` on the Bootstrap node are in the
+     storage volumes for both Prometheus and Alertmanager have been successfully
+     provisioned.
+
+- If you encouter an error during **installation** or have difficulties
```
Not sure the bold font is useful here.

Ebaneck committed:
docs: Enhance Bootstrap node deployment docs

This commit adds missing information to the bootstrap section
of MetalK8s documentation.

This commit adds:
Better explanation of the bootstrap configuration file
Describes the possible combinations of proxy and IPIP
Link to troubleshooting guide
Move of prometheus volume provisioning to post-install

Closes: #2114

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,178 @@
+
+- Check and ensure that the **MetalK8s ISO** is mounted properly.
+
+  .. code-block:: shell
```
You need to use `parsed-literal` for substitutions like `|release|` to work

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,178 @@
+If after provisioning a Volume for a Pod (e.g. Prometheus) and the PVC still
+hangs in a ``Pending`` state, then try checking the following:
+
+- Check that the volumes have been provisioned and it's status is ``Ready``:
```
```suggestion
- Check that the volumes have been provisioned and are in a **Ready** state:
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,178 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If after provisioning a Volume for a Pod (e.g. Prometheus) and the PVC still
+hangs in a ``Pending`` state, then try checking the following:
```
```suggestion
hangs in a **Pending** state, then try checking the following:
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,178 @@
+Access to MetalK8s GUI fails with "undefined backend"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If in the cause of using the MetalK8s GUI, you encounter an "undefined
+backend" error then perform the following checks.
```
```suggestion
backend" error then perform the following checks:
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,178 @@
+If after installation of a MetalK8s cluster you notice that Pod-to-Pod
+communication has routing problems, perform the following:
+
+- Check the configured values for the internal POD and Service networks:
```
```suggestion
- Check the configured values for the internal Pod and Service networks:
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,178 @@
+         workload_plane:
+             172.21.254.32/27
+
+  Make sure the configured IP ranges (CIDR notation) do not conflicts with your
```
```suggestion
  Make sure the configured IP ranges (CIDR notation) do not conflict with your
```

gdemonet reviewed (changes requested):
Last changes and it'll be all good!

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/2114-enhance-bootstrap-node-deployment-documentation` and the
destination branch `development/2.4`.

Please resolve the conflict on **the feature branch** (`improvement/2114-enhance-bootstrap-node-deployment-documentation`).

```bash
 $ git fetch
 $ git checkout origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/2114-enhance-bootstrap-node-deployment-documentation
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

Ebaneck commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

Ebaneck commented:
/reset

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

Ebaneck commented:
/status

bert-e commented:
# Status





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

Ebaneck commented:
/help

bert-e commented:
# Help page



The following options and commands are available at this time.

## Options ##


name   | description  | privileged   | authored
------ | ------------ | ------------ |----------
:arrow_right: **create_pull_requests** | Allow the creation of integration pull requests. |  | 
:arrow_right: **bypass_author_approval** | Bypass the pull request author's approval |  :star:  | 
:arrow_right: **bypass_commit_size** | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
:arrow_right: **unanimity** | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
:arrow_right: **bypass_incompatible_branch** | Bypass the check on the source branch prefix |  :star:  | 
:arrow_right: **bypass_build_status** | Bypass the build and test status |  :star:  | 
:arrow_right: **approve** | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
:arrow_right: **after_pull_request** | Wait for the given pull request id to be merged before continuing with the current one. |  | 
:arrow_right: **wait** | Instruct Bert-E not to run until further notice. |  | 
:arrow_right: **bypass_jira_check** | Bypass the Jira issue check |  :star:  | 
:arrow_right: **no_octopus** | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
:arrow_right: **bypass_leader_approval** | Bypass the pull request leaders' approval |  :star:  | 
:arrow_right: **bypass_peer_approval** | Bypass the pull request peers' approval |  :star:  | 



## Commands ##


name   | description  | privileged
------ | ------------ | ------------
:arrow_right: **reset** | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
:arrow_right: **help** | Print Bert-E's manual in the pull request. | 
:arrow_right: **build** | Re-start a fresh build ```TBA``` | 
:arrow_right: **force_reset** | Delete integration branches & pull requests, and restart merge process from the beginning. | 
:arrow_right: **retry** | Re-start a fresh build ```TBA``` | 
:arrow_right: **status** | Print Bert-E's current status in the pull request ```TBA``` | 
:arrow_right: **clear** | Remove all comments from Bert-E from the history ```TBA``` |

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

Ebaneck commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

* @gdemonet

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -27,8 +27,11 @@ Configuration
-   IP address, and hostname to conform to your infrastructure.
+#. Create the :file:`/etc/metalk8s/bootstrap.yaml` file.
+   This file contains initial configuration settings which are mandatory for
+   settings up a MetalK8s :term:`Bootstrap node`.
```
Whoops, missed that one:

```suggestion
   setting up a MetalK8s :term:`Bootstrap node`.
```

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -123,29 +168,18 @@ Bootstrap node.
+Validate the install
+^^^^^^^^^^^^^^^^^^^^
+- Check that all :term:`Pods <Pod>` on the Bootstrap node are in the
+  ``Running`` state. Note that Prometheus and Alertmanager pods will remain in
```
From what I've seen in @lucieleonard 's PR, user-facing values such as an object's `state` are shown as bold, not monospace:

```suggestion
  **Running** state. Note that Prometheus and Alertmanager pods will remain in
```

gdemonet commented on docs/installation/bootstrap.rst:
```diff
@@ -123,29 +168,18 @@ Bootstrap node.
+^^^^^^^^^^^^^^^^^^^^
+- Check that all :term:`Pods <Pod>` on the Bootstrap node are in the
+  ``Running`` state. Note that Prometheus and Alertmanager pods will remain in
+  a ``Pending`` state until their respective persistent storage volumes are
```
```suggestion
  a **Pending** state until their respective persistent storage volumes are
```

gdemonet commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,178 @@
+- If the ISO is unmounted, run the following command which will check the
+  the status of the ISO file and remount it automatically.
+
+  .. code-block:: shell
```
```suggestion
  .. parsed-literal::
```

gdemonet reviewed (changes requested):
Almost :innocent:

slaperche-scality commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,178 @@
+          Status:                True
+          Type:                  Ready
+
+- Check that a corresponding PersistentVolume exist:
```
Another thing worth checking is that the PV matches the PVC constraints (size, labels, storage class).

bert-e commented:
# History mismatch



Merge commit #cfec0309aa4b3cee205b02811ae0ac58a43802ce on the integration branch
`w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` is merging a branch which is neither the current
branch `improvement/2114-enhance-bootstrap-node-deployment-documentation` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2114-enhance-bootstrap-node-deployment-documentation` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

Ebaneck commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

* @gdemonet

slaperche-scality commented on docs/operation/troubleshooting.rst:
```diff
@@ -0,0 +1,202 @@
+     <volume-name>              10Gi       RWO            Retain          Bound  <storage-class-name>     4d22h     <persistentvolume-claim-name>
+
+- Check that the PersistentVolume matches the PersistentVolume Claim
+  constraints(size, labels, storage class) by doing the following:
```
missing space before `(`

bert-e commented:
# History mismatch



Merge commit #16ffbba26d4db312b1b225e00607e7ea6b9ebd9a on the integration branch
`w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` is merging a branch which is neither the current
branch `improvement/2114-enhance-bootstrap-node-deployment-documentation` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2114-enhance-bootstrap-node-deployment-documentation` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

Ebaneck commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation` with contents from `improvement/2114-enhance-bootstrap-node-deployment-documentation`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation origin/development/2.5
 $ git merge origin/improvement/2114-enhance-bootstrap-node-deployment-documentation
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2114-enhance-bootstrap-node-deployment-documentation
```

Ebaneck commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

* @gdemonet







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/28e87783a44daec53726c10a4998f24945a3ea27) did not succeed in branch improvement/2114-enhance-bootstrap-node-deployment-documentation.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye ebaneck.
